gc/base/MemorySubSpace.cpp

```diff
@@ -1127,6 +1127,38 @@ MM_MemorySubSpace::canExpand(MM_EnvironmentBase* env, uintptr_t expandSize)
 	return false;
 }
 
+/**
+ * Compare the specified expand amount with -XsoftMX value
+ * @return Updated expand size
+ */
+uintptr_t
+MM_MemorySubSpace::adjustExpansionWithinSoftMax(MM_EnvironmentBase *env, uintptr_t expandSize, uintptr_t minimumBytesRequired, uintptr_t memoryType)
+{
+	MM_Heap *heap = env->getExtensions()->getHeap();
+
+	uintptr_t actualSoftMx = heap->getActualSoftMxSize(env, memoryType);
+	uintptr_t activeMemorySize = getActiveMemorySize();
+	OMRPORT_ACCESS_FROM_OMRPORT(env->getPortLibrary());
+
+	if (0 != actualSoftMx) {
+		if ((0 != minimumBytesRequired) && ((activeMemorySize + minimumBytesRequired) > actualSoftMx)) {
+			if (J9_EVENT_IS_HOOKED(env->getExtensions()->omrHookInterface, J9HOOK_MM_OMR_OOM_DUE_TO_SOFTMX)) {
+				ALWAYS_TRIGGER_J9HOOK_MM_OMR_OOM_DUE_TO_SOFTMX(env->getExtensions()->omrHookInterface, env->getOmrVMThread(), omrtime_hires_clock(),
+						heap->getMaximumMemorySize(), heap->getActiveMemorySize(), env->getExtensions()->softMx, minimumBytesRequired);
```

Comment:
it's probably better to report active memory size and softMx for the specific memory type, since it's what the values are used in comparions.   (admittedly, it would be also nice to report total softMx, too)